src/pages/ReimbursementAccount/NonUSD/BusinessInfo/BusinessInfo.tsx

```diff
@@ -22,14 +29,98 @@ type BusinessInfoProps = {
+    ANNUAL_VOLUME: INPUT_IDS.ADDITIONAL_DATA.CORPAY.ANNUAL_VOLUME,
+    TRADE_VOLUME: INPUT_IDS.ADDITIONAL_DATA.CORPAY.TRADE_VOLUME,
+    TAX_ID_EIN_NUMBER: INPUT_IDS.ADDITIONAL_DATA.CORPAY.TAX_ID_EIN_NUMBER,
+};
 
 function BusinessInfo({onBackButtonPress, onSubmit}: BusinessInfoProps) {
     const {translate} = useLocalize();
 
-    const submit = () => {
-        onSubmit();
-    };
+    const [reimbursementAccount] = useOnyx(ONYXKEYS.REIMBURSEMENT_ACCOUNT);
+    const [reimbursementAccountDraft] = useOnyx(ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM_DRAFT);
+    const policyID = reimbursementAccount?.achData?.policyID;
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`);
+    const currency = policy?.outputCurrency ?? '';
+    const onyxValues = useMemo(() => getSubstepValues(INPUT_KEYS, reimbursementAccountDraft, reimbursementAccount), [reimbursementAccount, reimbursementAccountDraft]);
+    const bankAccountID = reimbursementAccount?.achData?.bankAccountID ?? CONST.DEFAULT_NUMBER_ID;
+
+    const country = reimbursementAccount?.achData?.additionalData?.[INPUT_IDS.ADDITIONAL_DATA.COUNTRY] ?? reimbursementAccountDraft?.[INPUT_IDS.ADDITIONAL_DATA.COUNTRY] ?? '';
+
+    useEffect(() => {
+        BankAccounts.getCorpayOnboardingFields(country);
+    }, [country]);
+
+    const submit = useCallback(() => {
+        BankAccounts.saveCorpayOnboardingCompanyDetails(
+            {
+                annualVolume: onyxValues[INPUT_KEYS.ANNUAL_VOLUME],
+                applicantTypeId: onyxValues[INPUT_KEYS.APPLICANT_TYPE_ID],
+                companyName: onyxValues[INPUT_KEYS.NAME],
+                companyStreetAddress: onyxValues[INPUT_KEYS.STREET],
+                companyCity: onyxValues[INPUT_KEYS.CITY],
+                companyState: onyxValues[INPUT_KEYS.STATE],
+                companyPostalCode: onyxValues[INPUT_KEYS.ZIP_CODE],
+                companyCountryCode: onyxValues[INPUT_KEYS.COUNTRY],
+                currencyNeeded: currency,
+                businessContactNumber: onyxValues[INPUT_KEYS.CONTACT_NUMBER],
+                businessConfirmationEmail: onyxValues[INPUT_KEYS.CONFIRMATION_EMAIL],
+                businessRegistrationIncorporationNumber: onyxValues[INPUT_KEYS.BUSINESS_REGISTRATION_INCORPORATION_NUMBER],
+                formationIncorporationState: onyxValues[INPUT_KEYS.INCORPORATION_STATE],
+                formationIncorporationCountryCode: onyxValues[INPUT_KEYS.INCORPORATION_COUNTRY],
+                fundSourceCountries: onyxValues[INPUT_KEYS.COUNTRY],
+                fundDestinationCountries: onyxValues[INPUT_KEYS.STATE],
+                natureOfBusiness: onyxValues[INPUT_KEYS.BUSINESS_CATEGORY],
+                purposeOfTransactionId: CONST.NON_USD_BANK_ACCOUNT.PURPOSE_OF_TRANSACTION_ID,
+                tradeVolume: onyxValues[INPUT_KEYS.TRADE_VOLUME],
+                taxIDEINNumber: onyxValues[INPUT_KEYS.TAX_ID_EIN_NUMBER],
```

Comment:
I'm fine with this way but I think we can shorten it:
Here's the idea, lmk what do you think 😄 
```javascript
    const params = {};

    Objects.key(INPUT_KEYS).forEach((currentKey) => {
        params[currentKey] = onyxValues[currentKey];
    });

    BankAccounts.saveCorpayOnboardingCompanyDetails({...params, currencyNeeded: currency})

```

Reply:
I think it looks better. Let's do it 😄

Reply:
Updated